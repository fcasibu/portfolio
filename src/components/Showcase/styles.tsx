import styled, { keyframes } from 'styled-components';

const fadeIn = keyframes`
  from {
    padding-top: 11rem;
    opacity: 0;
  }

  to {
    opacity: 1;
    padding-top: 7rem;
  }
`;

export const ShowcaseWrapper = styled.div`
  display: flex;
  flex-direction: column;
  align-items: center;
  text-align: center;
  margin-bottom: 11rem;
  animation: ${fadeIn} 1s ease-in-out forwards;
`;

export const ShowcaseTitle = styled.h1`
  font-size: 4rem;
  line-height: 1;
`;

export const ShowcaseDescription = styled.p`
  font-size: 2rem;
  margin-bottom: 1rem;
`;
