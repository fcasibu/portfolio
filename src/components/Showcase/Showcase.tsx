import Button from '../Button';
import { ShowcaseDescription, ShowcaseTitle, ShowcaseWrapper } from './styles';

interface Props {
  scrollToProjects(): void;
}

const Showcase = ({ scrollToProjects }: Props) => {
  return (
    <ShowcaseWrapper>
      <ShowcaseTitle>Hi, I&apos;m Francis</ShowcaseTitle>
      <ShowcaseDescription>I&apos;m a full stack web developer</ShowcaseDescription>
      <Button size="medium" onClick={scrollToProjects}>
        View my projects
      </Button>
    </ShowcaseWrapper>
  );
};

export default Showcase;
