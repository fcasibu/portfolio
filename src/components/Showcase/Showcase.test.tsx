import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import Showcase from './Showcase';

describe('Showcase', () => {
  it('should render and match the snapshot', () => {
    const { container } = render(<Showcase scrollToProjects={jest.fn()} />);
    expect(container).toMatchSnapshot();
  });

  it('should render the elements properly', () => {
    render(<Showcase scrollToProjects={jest.fn()} />);
    const title = screen.getByRole('heading', { name: /hi, i'm francis/i });
    const description = screen.getByText(/i'm a full stack web developer/i);
    const button = screen.getByRole('button', { name: /view my projects/i });

    expect(title).toBeInTheDocument();
    expect(description).toBeInTheDocument();
    expect(button).toBeInTheDocument();
  });
});
