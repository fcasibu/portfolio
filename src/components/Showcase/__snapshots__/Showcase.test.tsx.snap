// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Showcase should render and match the snapshot 1`] = `
<div>
  <div
    class="sc-gsnTZi iHWRiE"
  >
    <h1
      class="sc-dkzDqf hFzkqp"
    >
      Hi, I'm Francis
    </h1>
    <p
      class="sc-hKMtZM gaZiMG"
    >
      I'm a full stack web developer
    </p>
    <button
      class="sc-bczRLJ hsREaV"
    >
      View my projects
    </button>
  </div>
</div>
`;
